src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+    tooltipContent.appendChild(
+      createTooltipNode({
+        tag: result.resource
+          ? `${result.resource?.[0]}/${result.resource?.[1]}`
+          : null,
+        title: result.title ?? result._self,
+        isDownloadable: result.isDownloadable,
+      })
+    );
+    return tooltipContent;
+  }
+  if (resolvedAs === 'resources') {
+    tooltipContent.appendChild(
+      createTooltipNode({
+        tag: 'Multiple',
+        title: `${
+          (results as TDELink[]).length
+        } resources was found, click to list them`,
+      })
+    );
+    return tooltipContent;
+  }
+  if (resolvedAs === 'external') {
+    tooltipContent.appendChild(
+      createTooltipNode({
+        tag: 'External',
+        title: (results as TDELink).title ?? (results as TDELink)._self,
+      })
+    );
+    return tooltipContent;
+  }
+  return null;
+}
+
+function createPopoverContent({
+  results,
+  onClick,
+}: {
+  results: TDELink[];
+  onClick: (result: TDELink) => void;
+}) {
+  const tooltipContent = document.createElement('div');
+  tooltipContent.className = clsx(
+    `${CODEMIRROR_HOVER_CLASS}-resources-content`
+  );
+  // create node for each link in results and then append it to the tooltipContent
+  (results as TDELink[]).forEach((link: TDELink) => {
+    const linkNode = createTooltipNode({
+      tag: link.resource ? `${link.resource?.[0]}/${link.resource?.[1]}` : null,
+      title: link.title ?? link._self,
+      isDownloadable: link.isDownloadable,
+    });
+    linkNode.onclick = () => {
+      removePopoversFromDOM();
+      onClick(link);
+    };
+    return tooltipContent.appendChild(linkNode);
+  });
+  return tooltipContent;
+}
+function useEditorTooltip({
+  ref,
+  isEditing,
+  orgLabel,
+  projectLabel,
+}: {
+  ref: React.MutableRefObject<CodeMirror.Editor | undefined>;
+  isEditing: boolean;
+  orgLabel: string;
+  projectLabel: string;
+}) {
+  const nexus = useNexusContext();
+  const {
+    config: { apiEndpoint },
+  } = useSelector((state: RootState) => ({
+    config: state.config,
+  }));
+
+  const allowTooltip = !isEditing;
+
+  React.useEffect(() => {
+    const currentEditor = (ref as React.MutableRefObject<CodeMirror.Editor>)
+      ?.current;
+    const editorWrapper = currentEditor.getWrapperElement();
+    function positionner(ev: MouseEvent, tooltip: HTMLDivElement) {
+      const editorRect = (ev.target as HTMLElement).getBoundingClientRect();
+      const tooltipRect = tooltip.getBoundingClientRect();
+      if (tooltipRect.height <= editorRect.top) {
+        tooltip.style.top = `${editorRect.top - tooltipRect.height}px`;
+      } else {
+        tooltip.style.top = `${editorRect.bottom}px`;
+      }
+      tooltip.style.left = `${editorRect.left}px`;
+    }
+    function removeTooltipFromDom(tooltip: HTMLDivElement) {
+      if (tooltip.parentNode) {
+        tooltip.parentNode.removeChild(tooltip);
+      }
+    }
+    function hideTooltip(tooltip: HTMLDivElement) {
```

Comment:
I find the abstractions in this useEffect somewhat confusing. We have 3 functions:
i) `hideTooltip`
ii) `removeTooltip`
iii) `hide`

They all sound like they should be hiding a tooltip so it's confusing why we need 3 of these. Most certainly we can combine `hideTooltip` and `removeTooltipFromDom` into one function (otherwise the abstraction seems too shallow, I'm afraid). 
The last function `hide` maybe does more than just hide the tooltip, so maybe a better name could be `cleanup`?

Reply:
yeah, it was just a mess up trying to refactor removing from DOM also for popover,